### PR TITLE
create documentation for active directory configuration for OSA clusters

### LINKED ISSUE #50: document AAD configuration

### PR DESCRIPTION
**Pull request by charlesakalugwu**
closes #50

### CODE REVIEW

**Comment by jim-minter** [comment]
@charlesakalugwu please implement or close :)

File: docs/active_directory_configuration.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+- At the top of the page, type `Azure Active Directory` and click the search result to visit the Azure Active Directory Service page
+- Click on `App Registrations`
+- Click `New application registration`
+	- Type a unique name. A good value to use here is the value that was/will be passed to `hack/create.sh` during OSA cluster creation.
+	- Application type should remain `Web app /API`
+	- Sign-on URL should be `https://openshift.<cluster_name>.<dns_suffix>/oauth2callback/Azure%20AD` where:
+		- `<cluster_name>` is the same argument that was/will be passed to hack/create.sh  during OSA cluster creation and 
+		- `<dns_suffix>` is the value of the DNS_DOMAIN from the env file that was/will be used during OSA cluster creation.
+- Click `Create`
+- The value of `Application ID` will be used as the value of `AZURE_AAD_CLIENT_ID` in the env file used for OSA cluster creation
```
Comment (mjudeikis) [inline docs/active_directory_configuration.md]: Should we state that it is user responsibility to set it?

File: docs/active_directory_configuration.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+- Click `Settings`
+- Click `Keys`
+	- In the form within the `Passwords` section, type a description, select an expiration date and click the save button. Copy the text displayed in the `Value` input field and use that as the value of `AZURE_AAD_CLIENT_SECRET` in the env file used for OSA cluster creation
```
Comment (mjudeikis) [inline docs/active_directory_configuration.md]: I think we dont need this. If `AZURE_AAD_CLIENT_ID` is set, `aad.sh` is invoked and the random secret is generated and callback url updated automatically. This means that you can reuse the same application multiple times.

File: docs/active_directory_configuration.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	- This brings you to the `Select permissions` dialog. Select the following combinations of api/role mappings:
+		- Application permissions
+			- Read all groups
+			- Read directory data
+		- Delegated permissions
+			- Read all groups
+			- Sign users in
+	- Click save
+	- Copy the current azure portal page link from the browser and ask an admin to grant consent for your AAD application to assume those roles
+	- After this you should have your env file in the root of the openshift-azure project configured with the `AZURE_AAD_CLIENT_ID` and `AZURE_AAD_CLIENT_SECRET` retrieved from the steps above.
```
Comment (mjudeikis) [inline docs/active_directory_configuration.md]: Same as above. The secret is generated by `aad.sh`

File: docs/active_directory_configuration.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+grants access to the OSA cluster components, enabling them to perform 
+CRUD operations against the Azure Resource Provider API. The credentials 
+for the Service Provider principal are specified in `AZURE_CLIENT_ID` and 
+`AZURE_CLIENT_SECRET` in the env file which is sourced prior to `hack/create.sh` invocation.
+
+Every OSA cluster requires a Service Principal application. An additional AAD 
+enterprise application could be configured if the cluster operator intends to 
+enable integration of the OSA cluster with AAD authentication flows. 
+
+### Configuring your own AAD for an OpenShift Cluster on Azure
```
Comment (mjudeikis) [inline docs/active_directory_configuration.md]: All this section cloud become more abstract and create.sh is explained below

File: docs/active_directory_configuration.md (hunk 2 above)
Comment (charlesakalugwu) [reply docs/active_directory_configuration.md]: alright. i'll update this.

File: docs/active_directory_configuration.md (hunk 3 above)
Comment (charlesakalugwu) [reply docs/active_directory_configuration.md]: alright. i'll update this.

File: docs/active_directory_configuration.md (hunk 1 above)
Comment (charlesakalugwu) [reply docs/active_directory_configuration.md]: it is stated in line 48 below

File: docs/active_directory_configuration.md (hunk 4 above)
Comment (charlesakalugwu) [reply docs/active_directory_configuration.md]: I explained a bit more about using create.sh at the end.... in connection with the information retrieved from performing the preceeding `aad` config steps.

**Comment by charlesakalugwu** [comment]
Thanks for the link! I'll split up the doc, move them to openshift-azure and incorporate information from the link.

**Comment by charlesakalugwu** [comment]
moving this to openshift-azure

**Pull request closed without merging**